File: 7-kyu/simple-maths-test/javascript/solution.js
const isEven = n => n % 2 === 0;

const divisibleByTen = n => n % 10 === 0;

const isPrime = n => {
  if (n === 2 || n === 3) {
    return true;
  }
  if (n < 2 || n % 2 === 0) {
    return false;
  }
  const limit = Math.trunc(Math.sqrt(n));
  for (let i = 3; i <= limit; i += 2) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
};

const numberProperty = n => {
  const prime = isPrime(n);
  const even = isEven(n);
  const ten = divisibleByTen(n);
  return [prime, even, ten];
};

File: 7-kyu/simple-maths-test/python/solution.py
def is_prime(n):
    if n == 2 or n == 3:
        return True
    if n < 2 or n % 2 == 0:
        return False
    for i in range(3, int(n ** 0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

def number_property(n):
    prime = is_prime(n)
    even = n % 2 == 0
    ten = n % 10 == 0
    return [prime, even, ten]